src/net/server/handlers/query.rs

```diff
@@ -64,13 +65,36 @@ impl<T> Clone for ReservedPermit<T> {
 
 /// Middleware that first reserves a permit on the channel to send messages to the messaging layer.
 /// Once reserved, adds the permit to the extension for retrieval from the handler.
-pub async fn obtain_permit_mw<T: Send + 'static, B>(
-    sender: mpsc::Sender<T>,
-    mut req: Request<B>,
+/// # Panics
+/// if messages arrive out of order
+pub async fn obtain_permit_mw<B: Send>(
+    req: Request<B>,
     next: Next<B>,
 ) -> Result<Response, MpcHelperServerError> {
+    let mut req_parts = RequestParts::new(req);
+    let Path(query_id, step) = Path::from_request(&mut req_parts).await?;
+    let Query(RoleQueryParam { role }) =
+        Query::<RoleQueryParam>::from_request(&mut req_parts).await?;
+    let record_headers = RecordHeaders::from_request(&mut req_parts).await?;
+    let Extension(last_seen_messages) =
+        Extension::<LastSeenMessages>::from_request(&mut req_parts).await?;
+    // PANIC if messages arrive out of order
```

Comment:
I wonder if you validated this behavior somehow. Not insisting on writing a unit test on a piece of functionality that will be gone soon, but it would be good to have some level of confidence that it will work as expected. I'll be happy with manual test if that is easier to do

Reply:
yeah, i did verify that this worked. I had to modify the tests to get them to pass.

for instance, in the `backpressure_applied` test, if you comment out the `offset += 1`, the test panics with
```
called `Result::unwrap()` on an `Err` value: OutOfOrder { last_seen: 1, next_seen: 0 }
thread 'net::server::handlers::query::tests::backpressure_applied' panicked at 'called `Result::unwrap()` on an `Err` value: OutOfOrder { last_seen: 1, next_seen: 0 }', src/net/server/handlers/query.rs:85:10
```

Reply:
great to hear! if it is a matter of single line change, can we add a test for it?

Reply:
done